kedro-datasets/kedro_datasets/spark/spark_dataset.py

```diff
@@ -26,6 +27,27 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_spark() -> Any:
+    """
+    Returns the SparkSession, we need this wrapper because the SparkSession
+    is retrieved differently in databricks-connect vs a Notebook in web UI
+    """
+    try:
+        # When using databricks-connect >= 13.0.0 (a.k.a databricks-connect-v2)
+        # the remote session is instantiated using the databricks module 
+        # If the databricks-connect module is installed, we use a remote session 
+        from databricks.connect import DatabricksSession
+
+        spark = DatabricksSession.builder.getOrCreate()
+
+    except ImportError:
+        # For "normal" spak sessions that don't use databricks-connect
+        # we get spark normally
+        spark = SparkSession.builder.getOrCreate()
```

Comment:
For even further improved compatibility, add
```
from pyspark.sql.connect.session import SparkSession as RemoteSparkSession
```

and replace this line with `RemoteSparkSession.builder.getOrCreate()`.

Reply:
Thanks a lot for looking into this @cdkrot , super appreciated

Do you mean another try except? So the fallback waterfall would be:

1. DatabricksSession
2. RemoteSparkSession
3. SparkSession

I guess adding RemoteSparksession would helo adding support for non-Databricks Spark-connect sessions?

Reply:
Sorry, my bad, the logic in code is good as is 👍. sql.SparkSession will figure out if it needs to create sql.SparkSession or connect.SparkSession.

Possibly it's best to rewrite the comments though. Essentially there is no reason to fallback from dbconnect session creation to pyspark's since dbconnect does it automatically, the only reason to fallback is if dbconnect is not installed.